plugins/modules/aci_config_rollback.py

```diff
@@ -291,17 +293,32 @@ def get_preview(aci):
     This function is used to generate a preview between two snapshots and add the parsed results to the aci module return data.
     '''
     uri = aci.url + aci.filter_string
-    resp, info = fetch_url(aci.module, uri, headers=aci.headers, method='GET', timeout=aci.module.params.get('timeout'),
-                           use_proxy=aci.module.params.get('use_proxy'))
+    path = aci.path + aci.filter_string
+    resp = None
+    if aci.module._socket_path:
+        conn = Connection(aci.module._socket_path)
+        conn.set_auth(aci.headers.get('Cookie'), aci.module.params.get('host'), aci.module.params.get('username'),
+                      aci.module.params.get('password'), aci.module.params.get('port'), aci.module.params.get('use_ssl'),
+                      aci.module.params.get('use_proxy'), aci.module.params.get('validate_certs'))
+        info = conn.send_request('GET', '/{0}'.format(path))
+    else:
+        resp, info = fetch_url(aci.module, uri, headers=aci.headers, method='GET', timeout=aci.module.params.get('timeout'),
+                               use_proxy=aci.module.params.get('use_proxy'))
```

Comment:
Any reason not to call aci.call() ?

Reply:
The output for our GET method in aci.py has certain elements which are not present in get_preview's output when
these are used:
aci.url = '%(protocol)s://%(host)s/mqapi2/snapshots.diff.xml' % module.params
        aci.path = 'mqapi2/snapshots.diff.xml'
        aci.filter_string = (
            '?s1dn=uni/backupst/snapshots-[uni/fabric/configexp-%(export_policy)s]/snapshot-%(snapshot)s&'
            's2dn=uni/backupst/snapshots-[uni/fabric/configexp-%(compare_export_policy)s]/snapshot-%(compare_snapshot)s'
        ) % module.params